docker/ci/Dockerfile-clang

```diff
@@ -0,0 +1,37 @@
+FROM ubuntu:16.04
+
+ENV DEBIAN_FRONTEND=noninteractive
+
+RUN apt-get update -qq && apt-get install -yqq \
+    build-essential \
+    cmake \
+    g++ \
```

Comment:
is g++ required for a clang build?

Reply:
This was left over from the previous docker image, requirements have been adjusted for clang specifically